scripts/version-sync.sh

```diff
@@ -108,11 +108,18 @@ for i in "${!FILES_TO_CHECK[@]}"; do
     RE_SEMVER=${RE_SEMVERS[$i]}
     UPDATED_VERSION=${UPDATED_VERSIONS[$i]}
     FILE_VERSION=$(grep -o -m 1 -E "${RE_SEMVER}" "$FILE_TO_CHANGE")
+    CURRENT_RELEASE=$(curl -sL https://api.github.com/repos/Unstructured-IO/unstructured/releases/latest | jq -r ".tag_name")
```

Comment:
rather than using the github API, please use the git command line.
e.g., before the loop begins, you can define:

```
MAIN_VERSION=$(git show main:unstructured/__version__.py | grep -o -m 1 -E "${RE_SEMVER_FULL}")
```

Reply:
actually, also need to check if that version is not a dev version but a real release. could do that with another env var:
```
MAIN_VERSION=$(git show main:unstructured/__version__.py | grep -o -m 1 -E "${RE_SEMVER_FULL}")
MAIN_IS_RELEASE=false
[[ $MAIN_VERSION != *"-dev"* ]] && MAIN_IS_RELEASE=true
```

Reply:
thanks for the feedback - didn't realize I could use the command line here :) 

By using the github api call, I knew I would be getting the latest release, not a dev version. In your suggestion, by checking the __version__.py file, we then have to add the check for it being a dev version, but I thought the point was to make sure that the proposed release version is not the same as the latest release, whether or not there have been subsequent dev versions worked on.

For example, if the latest release is 0.8.8, and there has since been a 0.8.9-dev1 (let's say that's what's currently in __version__.py), and the proposed new release version in a new PR is 0.8.8 (same as the current release, which is the case we want to catch), if we only look at the version in __version__.py file (in this example 0.8.9-dev1), we would ignore the whole check against the current release and not realize that the PR is proposing the same release version as the current. The only other way this wouldn't be the case is if there is some sort of check for chronological versions (i.e., 0.8.8 cannot come after 0.8.9-dev1), but I don't believe that is currently in place.

I'm not saying we have to use the github api, cli is probably fine, but I do think we need the latest actual release and not just the version in __version__.py since we want to be comparing actual releases and not involve dev versions. The api was nice in that we could request the latest release tag and know that it won't be a dev version. How else might we find this information on main without the API?

Reply:
> For example, if the latest release is 0.8.8, and there has since been a 0.8.9-dev1 (let's say that's what's currently in version.py), and the proposed new release version in a new PR is 0.8.8 (same as the current release, which is the case we want to catch), 

Actually, i had thought. the primary case we want to catch is two `0.8.8`'s being committed in a row. I.e., it is much more likely to have an accidental follow on commit to a release rather than going backwards in CHANGELOG. 

More generally, there could (should) also be a separate semver check to ensure the CHANGELOG version >= current version. (super simple to do with say https://pypi.org/project/semver/ , not super simple in bash). I.e., one also shouldn't go from `-dev3` to `-dev2`.

The other problem of relying on the github API is human's need to generate releases, and once in awhile they forget to 😐 .

Reply:
That makes more sense, thanks for the clarification! 

So it seems like to address the primary case:
> an accidental follow on commit to a release

the changes you proposed using the git cli to get the version from `__version__.py` in main are best for now, and then I can make a new ticket for the general case of checking for an increment in version (whether dev or numerical)